ivy_models/mlpmixer/mlpmixer.py

```diff
@@ -0,0 +1,111 @@
+import ivy
+from huggingface_hub import hf_hub_download
+import joblib
+
+
+class MLPBlock(ivy.Module):
+    def __init__(self, inp_dim, mlp_dim):
+        self.linear0 = ivy.Linear(inp_dim, mlp_dim)
+        self.gelu = ivy.GELU()
+        self.linear1 = ivy.Linear(mlp_dim, inp_dim)
+        self.dropout = ivy.Dropout(0.25)
+        super(MLPBlock, self).__init__()
+
+    def _forward(self, x):
+        y = self.linear0(x)
+        y = self.gelu(y)
+        # y = self.dropout(y)
+        y = self.linear1(y)
+        # y = self.dropout(y)
+        return y
+
+
+class MixerBlock(ivy.Module):
+    def __init__(self, hidden_dim, num_patches, token_mlp_dim, channel_mlp_dim):
+        self.norm1 = ivy.LayerNorm([hidden_dim])
+        self.mlp_block1 = MLPBlock(num_patches, token_mlp_dim)
+        self.norm2 = ivy.LayerNorm([hidden_dim])
+        self.mlp_block2 = MLPBlock(hidden_dim, channel_mlp_dim)
+        super(MixerBlock, self).__init__()
+
+    def _forward(self, x):
+        y = self.norm1(x)
+        y = ivy.matrix_transpose(y)
+        y = self.mlp_block1(y)
+        y = ivy.matrix_transpose(y)
+        x = x + y
+        y = self.norm2(x)
+        y = x + self.mlp_block2(y)
+        return y
+
+
+class MLPMixer(ivy.Module):
+    def __init__(
+        self,
```

Comment:
there's some general conventions to follow, more info in this guide! 
ref: https://docs.google.com/document/d/1NfVngAQlQQlxLGEg47O2lnfn60M5xZekZFei9k5RJd8/edit?usp=sharing